src/promptflow/promptflow/_sdk/operations/_run_operations.py

```diff
@@ -138,13 +140,20 @@ def stream(self, name: Union[str, Run]) -> Run:
             available_logs = local_storage.logger.get_logs()
             incremental_print(available_logs, printed, file_handler)
             self._print_run_summary(run)
-            # print error message when run is failed
-            if run.status == RunStatus.FAILED:
-                error_message = local_storage.load_exception()["message"]
-                print_red_error(error_message)
         except KeyboardInterrupt:
             error_message = "The output streaming for the run was interrupted, but the run is still executing."
             print(error_message)
+
+        if run.status == RunStatus.FAILED or run.status == RunStatus.CANCELED:
+            if run.status == RunStatus.FAILED:
+                error_message = local_storage.load_exception().get("message", "Run fails with unknown error.")
+            else:
+                error_message = "Run is canceled."
+            if raise_on_error:
+                raise InvalidRunStatusError(error_message)
+            else:
+                print_red_error(error_message)
```

Comment:
What's current behavior when pf run create with stream == True? Will we print duplicated error message?

Reply:
I don't think there will be duplicate error message, and if the run failed, I think you will get an exception instead.